gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskTest.groovy

```diff
@@ -1,10 +1,159 @@
 import org.gradle.testfixtures.ProjectBuilder
-import spock.lang.Specification
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
 
-class CreateManifestTaskTest extends Specification {
+class CreateManifestTaskTest extends GradleTestSpec {
+
+    def jsonMapper = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(new KebabCaseStrategy())
+    def yamlMapper = new ObjectMapper(new YAMLFactory())
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(new KebabCaseStrategy())
+
+    File mavenRepo
+
+    def 'Fail on missing recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///{{mavenRepo}}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+            }
+        '''.stripIndent().replace("{{mavenRepo}}", mavenRepo.getAbsolutePath())
+
+        when:
+        BuildResult buildResult = run(':testCreateManifest').buildAndFail()
+
+        then:
+        buildResult.task(':testCreateManifest').outcome == TaskOutcome.FAILED
+        buildResult.output.contains("The following products are recommended as dependencies but do not appear in the " +
+                "product dependencies or product dependencies ignored list: [group:name2, group:name]")
+    }
+
+    def 'Can ignore recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///{{mavenRepo}}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+                ignoredProductIds = [
+                    new com.palantir.gradle.dist.ProductId("group:name"), 
```

Comment:
we could, but it appears the convention of this code base is to have the task take the typed things and do the conversion in the extension - happy to change this though